infra/lambda/datasets-endpoint/test_function.py

```diff
@@ -0,0 +1,307 @@
+"""
+Basic Lambda function tests. Working Data Lake AWS environment is required
+(run '$ cdk deploy' before running tests).
+"""
+
+import logging
+import re
+import uuid
+
+import function  # pylint:disable=import-error
+import pytest
+
+logging.basicConfig(level=logging.INFO)
+logger = logging.getLogger(__name__)
+
+
+def test_request_no_method():
+    """Try request not containing method."""
+
+    body = {}
+
+    resp = function.lambda_handler({"body": body}, "context")
+    logger.info(resp)
+
+    assert resp["statusCode"] == 400
+    assert resp["body"]["message"] == "Bad Request: 'httpMethod' is a required property."
+
+
+def test_request_no_body():
+    """Try request not containing body."""
+
+    method = "POST"
+
+    resp = function.lambda_handler({"httpMethod": method}, "context")
+    logger.info(resp)
+
+    assert resp["statusCode"] == 400
+    assert resp["body"]["message"] == "Bad Request: 'body' is a required property."
+
+
+def test_post_method(db_truncate):  # pylint:disable=unused-argument
+    """Test Dataset creation using POST method."""
+
+    method = "POST"
+    body = {}
+    body["type"] = "RASTER"
+    body["title"] = "Dataset {}".format(uuid.uuid4().hex[:8])
```

Comment:
If we can use UUIDs (without truncation) in the test I would suggest not truncating the DB, since collisions are enormously unlikely.

Reply:
`uuid` was removed in test in 6c2cbedb451a36b9889ae89ef3fcd08454c044c0